force-app/main/default/classes/Security Recipes/Safely.cls

```diff
@@ -0,0 +1,159 @@
+/**
+ * @description Class wraps DML Calls in FLS / Crud checks. Library is baseed on
+ * a fluent api system. All calls are constructed, then chained with options.
+ * For instances. `new Safely().allOrNothing().doInsert(List<sObject>);`
+ *
+ * Notable chainable methods include:
+ * - allOrNothing() - this enforces the AllOrNothing DML flag. All DML is
+ *      eventually executed via Database.* methods which accept an allOrNothing
+ *      parameter requiring all of the records to succeed or fail.
+ * - throwIfRemovedFields() - this method, if called, will result in an
+ *     exception being thrown if any record being modified has fields removed
+ *     by the security decision.
+ */
+public with sharing class Safely {
+    @testVisible
+    private Boolean allOrNothing = false;
+    @testVisible
+    private Boolean throwIfRemovedFields = false;
+
+    public class RemovedFieldsException extends Exception {
+    }
+
+    public Safely allOrNothing() {
+        this.allOrNothing = true;
+        return this;
+    }
+
+    public Safely throwIfRemovedFields() {
+        this.throwIfRemovedFields = true;
+        return this;
+    }
+
+    /// Insert
```

Comment:
Question - Are we going to document these methods?

Reply:
good catch. done.